src/App.tsx

```diff
@@ -26,27 +26,137 @@ However, Slovakia’s Prime Minister Robert Fico said on Wednesday that the halt
 
 Fico has previously argued that the end of the deal would lead to higher gas and electricity prices in Europe, the news agency said.`;
 
+const originalWords = originalContent.split(/\s+/);
+
 export default function Home() {
   const contentRef = useRef<HTMLDivElement>(null);
+  const [currentWordIndex, setCurrentWordIndex] = useState(0);
+  const [typedWord, setTypedWord] = useState("");
+
+  useEffect(() => {
+    if (contentRef.current) {
+      contentRef.current.focus();
+    }
+  }, []);
+
+  const handleOnKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLDivElement>) => {
+      if (e.key === " " || e.key === "Enter") {
+        const isCorrect = originalWords[currentWordIndex] === typedWord;
+        if (isCorrect) {
+          setCurrentWordIndex((prev) => prev + 1);
+          setTypedWord("");
+        }
+      } else if (e.key === "Backspace") {
+        setTypedWord((prev) => prev.slice(0, -1));
+      } else if (e.key.length === 1) {
+        setTypedWord((prev) => prev + e.key);
+      }
+      e.preventDefault();
+    },
+    [currentWordIndex, typedWord],
+  );
+
+  /**
+   * Renders the typing test content with appropriate styling based on typing progress.
+   * - Words before the current word are black (fully typed correctly).
+   * - The current word is styled dynamically to indicate correct, incorrect, or pending letters.
+   * - Future words are gray (not yet typed).
+   */
+  const renderedContent = () => {
```

Comment:
Feeling nako do ned gyud ni i-separate na component personally lang for readability

```tsx
function Words(props: WordsProps) {
	// ...logic
	return props.words.map((word) => <Word word={word} />);
}

function Word({ bar, baz, foo }: { foo: boolean; bar: boolean; baz: boolean }) {
	// ...logic
	if (foo) return <TypedWord />;
	if (baz) return <CurrentWord />;
	if (baz) return <UnTypedWord />;
}
```